### ISSUE TITLE
About not setting Cookies automatically

### ISSUE THREAD

**Issue by tahv0** [Mastermindzh/react-cookie-consent#45]
https://github.com/Mastermindzh/react-cookie-consent/blob/f3c5bfcd83d9e38aa18715b3d897576001b1e0f0/src/index.js#L145

Could it be possible that setting cookies after replying would be optional? If one does a REST call (or other) after onAccept() and store the result in Db.

**Comment by Mastermindzh** [comment]
I don't think I get what you mean. 
If the user clicks "accept" and the acceptance code fires why would the cookie ever **not** be set?

If you simply want to store the accept in a DB I'd do the following (pseudo-code):

```
onAccept = () => {
	db.save('user', 'hasaccepted', true);
}
```
```
onDecline = () => {
	db.save('user', 'hasaccepted', false);
}
```

**Comment by tahv0** [comment]
My problem is that this component not only asks for a consent but also adds a cookie. I would like to ask for a permission only and `not` add a another cookie. acceptButton's onClick method always adds a cookie on the line I referenced above.

**Comment by Mastermindzh** [comment]
I don't know whether I want to support such an option just doesn't make sense to me and the workaround is trivial. 

I'll keep the issue open, see if others want this feature as well and adjust accordingly.

## Workaround:
Set the expires to 1 second in the options like so:
```js
var date = new Date();
date.setSeconds(date.getSeconds() + 1);

---
<CookieConsent expires={date}>
text
</CookieConsent>
```

## Why I dislike not settings a cookie
Say for example your user visits your page whilst not being logged in. 
If the only place you save your cookie settings is the database the user will have to log in in order to remove the cookie bar.

This means that every visit serves the user with the cookiebar unless they sign in.

In my opinion you should save the cookie as both a DB option and a cookie.
- If the user visits and the cookie is set -> don't show bar
- If the user visits, no cookie is set but the user is logged in -> get cookie value from DB and do stuff accordingly
- If the user visits, no cookie is set and the user is not logged in -> show cookie bar

## advanced workaround

Extend from CookieConsent and override the affected methods.

**Comment by tahv0** [comment]
I see your points. But how about using local storage?
Does decline() add a cookie as well? After asking approval it still sets a cookie even if user says 'no'?

**Comment by Mastermindzh** [comment]
Local storage can be used but is not exempt from the law:

> No - The law also affects anything that acts like a cookie, for example:
> 
> - Flash Cookies
> - HTML5 Local Storage
> 
> The ICO has said that it isn't good enough to just re-implement the tracking some other way outside of cookie storage.
[source](http://web.archive.org/web/20130326235146/http://www.theeucookielaw.com/)
---

Yes, when a user declines, the bar places a cookie to remember that...
That's actually pretty dumb of us! I'll rework the code tonight to not set a cookie on decline and write an upgrade guide from the current version to the new version.

Thanks for bringing this to my attention!

**Comment by Mastermindzh** [comment]
The bug is fixed in [2.4.0](https://www.npmjs.com/package/react-cookie-consent/v/2.4.0)

[Demo on StackBlitz](https://stackblitz.com/edit/react-cookie-consent-setdeclinebutton-example)

Don't forget to star if you like the project!

Have a nice day.
